client/lib/reader-feed-subscriptions/index.js

```diff
@@ -138,37 +135,18 @@ var FeedSubscriptionStore = {
 			return subscriptionTemplate.merge( sub );
 		} );
 
-		newSubscriptions = Immutable.List( subscriptionsWithState ); // eslint-disable-line new-cap
-
 		// Is it the last page?
 		if ( data.number === 0 ) {
 			isLastPage = true;
 		}
 
-		// Is it a new page of results?
-		if ( currentPage > 0 && data.page > currentPage ) {
-			combinedSubscriptions = currentSubscriptions.list.concat( newSubscriptions );
-
-			subscriptions = {
-				count: combinedSubscriptions.size,
-				list: combinedSubscriptions
-			};
-		} else {
-			// Looks like the first results we've received...
-			subscriptions = {
-				count: newSubscriptions.size,
-				list: newSubscriptions
-			};
-		}
+		forEach( subscriptionsWithState, function( subscription ) {
+			addSubscription( subscription.toJS(), false );
```

Comment:
The `toJS()` isn't ideal, but addSubscription() currently expects a plain JS object.